packages/teams-js/test/public/authentication.spec.ts

```diff
@@ -461,431 +511,744 @@ describe('Testing authentication module', () => {
+              resources: [mockResource],
+              claims: [mockClaim],
+              silent: false,
+              failureCallback: () => {
+                done();
+              },
+              successCallback: result => {
+                expect(result).toEqual('token');
+                done();
+              },
+            };
+
+            authentication.getAuthToken(authTokenRequest);
+
+            const message = framelessPostMock.findMessageByFunc('authentication.getAuthToken');
+            expect(message).not.toBeNull();
+            expect(message.args.length).toBe(3);
+            expect(message.args[0]).toEqual([mockResource]);
+            expect(message.args[1]).toEqual([mockClaim]);
+            expect(message.args[2]).toEqual(false);
+
+            framelessPostMock.respondToMessage({
+              data: {
+                id: message.id,
+                args: [true, 'token'],
+              },
+            } as DOMMessageEvent);
+          });
         });
 
-        it(`authentication.notifyFailure should do window redirect if callbackUrl is for win32 Outlook and auth failure happens from ${context} context`, async () => {
-          expect.assertions(2);
-          let windowAssignSpyCalled = false;
-          jest.spyOn(utils.mockWindow.location, 'assign').mockImplementation((url: string): void => {
-            windowAssignSpyCalled = true;
-            expect(url).toEqual(
-              `https://outlook.office.com/connectors?client_type=Win32_Outlook#/configurations&reason=${errorMessage}&authFailure`,
-            );
+        it(`authentication.getAuthToken should successfully return error from getAuthToken in case of failure in legacy flow from ${context} context`, done => {
+          framelessPostMock.initializeWithContext(context).then(() => {
+            const authTokenRequest = {
+              resources: [mockResource],
+              failureCallback: error => {
+                expect(error).toEqual(errorMessage);
+                done();
+              },
+              successCallback: () => {
+                done();
+              },
+            };
+
+            authentication.getAuthToken(authTokenRequest);
+
+            const message = framelessPostMock.findMessageByFunc('authentication.getAuthToken');
+            expect(message).not.toBeNull();
+            expect(message.args.length).toBe(3);
+            expect(message.args[0]).toEqual([mockResource]);
+            expect(message.args[1]).toBeNull();
+            expect(message.args[2]).toBeNull();
+            framelessPostMock.respondToMessage({
+              data: {
+                id: message.id,
+                args: [false, errorMessage],
+              },
+            } as DOMMessageEvent);
           });
+        });
 
-          await utils.initializeWithContext(context);
+        it(`authentication.getAuthToken should successfully return getAuthToken in case of success from ${context} context`, async () => {
```

Comment:
nit: this test case should probably be called "authentication.getAuthToken should successfully return auth token ...". (Same for the framed version of it.)

Reply:
Same for other test cases that return an auth token